content/docs/engine-dev/architecture/prefabs/_index.md

```diff
@@ -0,0 +1,648 @@
+| Patch           | Represents a JSON patch that is applied on top of the nested prefab instance to distinguish it from other nested instances that are instantiated from the same template (the source template). |
+| DOM             | Stands for Document Object Model. It is a language agnostic logical model that represents any object data in a tree-like hierarchical structure where each node in the tree holds some relevant information about the data. In prefab system, DOM values are represented by [RapidJSON](https://rapidjson.org/) objects. Prefab DOM types are defined in [`PrefabDomTypes`](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabDomTypes.h). |
+
+
+
+
+## Overview
+<!-- <a name="overview"> -->
+
+To get an understanding of what prefabs are, we first need to understand what entities are. Entities are the foundational game objects that encapsulate logic required to build game worlds. An entity is made up of components that define its properties. More information about entities and components can be found in [Guide to Component Development](/docs/user-guide/programming/components). 
+
+Prefabs serve the purpose of easily building and modifying large and complex worlds for games. Creating large worlds often requires duplicating a lot of entities in the world. While this can be done by simply creating a bunch of entities and duplicating them in the editor, modifying them becomes a pain point since you would have to go into each of those entities and make the same changes. You would also have to come up with custom tracking mechanisms to identify which entities to modify in which areas of the world. In addition, you may also want some entities to retain their old properties and not to change along with other entities. All of these problems are resolved by prefabs.
+
+Simply put, a prefab is a collection of entities and nested prefabs. The further sections will explain how this collection of entities and nested prefabs are used to solve the above problems and many more.
+
+
+### Prefab file format
+<!-- <a name="prefab-file-format"> -->
+
+Prefab files use JSON format. A prefab file has the following fields.
+
+Container Entity  
+: A wrapper entity, which is the parent of all other entities and nested prefabs contained within a prefab. This is also used to distinguish prefabs from regular entities in the <a href="/docs/user-guide/editor/entity-outliner/">Entity Outliner</a>. A prefab must have a single container entity.
+
+Entities
+: A container for entities owned by a prefab. A prefab may contain zero or more nested entities.
+
+Instances
+: A container for nested prefabs owned by a prefab. A prefab may contain zero or more nested prefabs.
+
+Source
+: The relative path to the prefab file that the nested prefab needs to load from.
+
+Patches
+: A list of changes that get applied on top of the nested prefab.
```

Comment:
No need to change this if you're okay with it, but should we in some way indicate that these fields are "inside" the instances?

Reply:
Good callout. I've rearranged this to indicate that 'source' and 'patches' are within 'instances'